### ISSUE TITLE
[TEST-FAILURE] MapReadQuorumTest failures (....ThrowsExceptionWhenQuorumSizeNotMet)

### ISSUE THREAD

**Issue by hasancelik** [hazelcast/hazelcast#5801]
```
java.lang.AssertionError: Expected exception: com.hazelcast.quorum.QuorumException
	at org.junit.internal.runners.statements.ExpectException.evaluate(ExpectException.java:32)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at com.hazelcast.test.AbstractHazelcastClassRunner$ThreadDumpAwareRunAfters.evaluate(AbstractHazelcastClassRunner.java:134)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at com.hazelcast.test.HazelcastSerialClassRunner.runChild(HazelcastSerialClassRunner.java:44)
	at com.hazelcast.test.HazelcastSerialClassRunner.runChild(HazelcastSerialClassRunner.java:26)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
```

https://hazelcast-l337.ci.cloudbees.com/job/Hazelcast-3.x-OpenJDK8/636/

**Comment by hasancelik** [comment]
https://hazelcast-l337.ci.cloudbees.com/job/Hazelcast-3.x-OpenJDK8-Quality-Outreach/com.hazelcast$hazelcast/147/testReport/junit/com.hazelcast.quorum/MapReadQuorumTest/testLocalKeySetOperationThrowsExceptionWhenQuorumSizeNotMet/

**Comment by hasancelik** [comment]
related issue:
#5992

**Comment by hasancelik** [comment]
https://hazelcast-l337.ci.cloudbees.com/job/Hazelcast-3.x-OpenJDK9-Quality-Outreach/167/#showFailuresLink

**Comment by hasancelik** [comment]
**CacheReadQuorumTest/testGetAllOperationThrowsExceptionWhenQuorumSizeNotMet**

https://hazelcast-l337.ci.cloudbees.com/job/Hazelcast-3.x-OpenJDK8/com.hazelcast$hazelcast/660/testReport/junit/com.hazelcast.quorum.cache/CacheReadQuorumTest/testGetAllOperationThrowsExceptionWhenQuorumSizeNotMet/

**Comment by eminn** [comment]
@mustafaiman can you have a look at the latest failure related to jcache ?

**Comment by mustafaiman** [comment]
@eminn have you made any change to solve https://github.com/hazelcast/hazelcast/issues/5992 . It seems like it disappeared. I am asking because cause of jcache failure may be similar to that one.

**Comment by hasancelik** [comment]
**CacheReadQuorumTest...**

https://hazelcast-l337.ci.cloudbees.com/job/Hazelcast-3.x-OpenJDK8/664/

**Comment by hasancelik** [comment]
```
java.lang.NullPointerException: null
	at java.util.Hashtable.put(Hashtable.java:466)
	at java.util.Properties.setProperty(Properties.java:160)
	at java.lang.System.setProperty(System.java:791)
	at com.hazelcast.client.quorum.HiDensityClientCacheQuorumTestSupport.killAllHazelcastInstances(HiDensityClientCacheQuorumTestSupport.java:107)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:504)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47)
```

**HiDensityCacheWriteQuorumTest**

https://hazelcast-l337.ci.cloudbees.com/job/Hazelcast-ee-OpenJDK9-Quality-Outreach/com.hazelcast$hazelcast-enterprise/15/testReport/junit/com.hazelcast.quorum/HiDensityCacheWriteQuorumTest/com_hazelcast_quorum_HiDensityCacheWriteQuorumTest/

**HiDensityClientCacheReadQuorumTest**

https://hazelcast-l337.ci.cloudbees.com/job/Hazelcast-ee-OpenJDK9-Quality-Outreach/com.hazelcast$hazelcast-enterprise-client/15/testReport/junit/com.hazelcast.client.quorum/HiDensityClientCacheReadQuorumTest/com_hazelcast_client_quorum_HiDensityClientCacheReadQuorumTest/

**Comment by hasancelik** [comment]
```
com.hazelcast.quorum.QuorumException: Cluster quorum failed, quorum minimum size: 3, current size: 2
	at com.hazelcast.quorum.impl.QuorumImpl.newQuorumException(QuorumImpl.java:137)
	at com.hazelcast.quorum.impl.QuorumImpl.ensureQuorumPresent(QuorumImpl.java:128)
	at com.hazelcast.quorum.impl.QuorumServiceImpl.ensureQuorumPresent(QuorumServiceImpl.java:119)
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.ensureQuorumPresent(OperationRunnerImpl.java:175)
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:153)
	at com.hazelcast.spi.impl.operationexecutor.classic.ClassicOperationExecutor.runOnCallingThread(ClassicOperationExecutor.java:366)
	at com.hazelcast.spi.impl.operationexecutor.classic.ClassicOperationExecutor.runOnCallingThreadIfPossible(ClassicOperationExecutor.java:334)
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.doInvokeLocal(Invocation.java:253)
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.doInvoke(Invocation.java:237)
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.invokeInternal(Invocation.java:204)
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.invoke(Invocation.java:174)
```

https://hazelcast-l337.ci.cloudbees.com/job/Hazelcast-ee-OpenJDK9-Quality-Outreach/com.hazelcast$hazelcast-enterprise-client-new/39/testReport/junit/com.hazelcast.client.quorum/HiDensityClientCacheReadQuorumTest/testContainsOperationSuccessfulWhenQuorumSizeMet/

**Comment by hasancelik** [comment]
https://hazelcast-l337.ci.cloudbees.com/job/Hazelcast-3.x/com.hazelcast$hazelcast-client/3801/testReport/junit/com.hazelcast.client.quorum.cache/ClientCacheReadWriteQuorumTest/testReplaceOperationSuccessfulWhenQuorumSizeMet/

**Comment by hasancelik** [comment]
https://hazelcast-l337.ci.cloudbees.com/job/Hazelcast-3.x/com.hazelcast$hazelcast-client/3893/testReport/junit/com.hazelcast.client.quorum.cache/ClientCacheReadWriteQuorumTest/testReplaceOperationSuccessfulWhenQuorumSizeMet/

**Comment by eminn** [comment]
Closing the issue since not failing recently...

**Comment by hasancelik** [comment]
**testGetAsyncOperationSuccessfulWhenQuorumSizeMet**
```
java.util.concurrent.ExecutionException: com.hazelcast.quorum.QuorumException: Cluster quorum failed, quorum minimum size: 3, current size: 2
	at com.hazelcast.quorum.impl.QuorumImpl.newQuorumException(QuorumImpl.java:140)
	at com.hazelcast.quorum.impl.QuorumImpl.ensureQuorumPresent(QuorumImpl.java:131)
	at com.hazelcast.quorum.impl.QuorumServiceImpl.ensureQuorumPresent(QuorumServiceImpl.java:118)
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.ensureQuorumPresent(OperationRunnerImpl.java:216)
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:164)
	at com.hazelcast.spi.impl.operationexecutor.classic.ClassicOperationExecutor.runOnCallingThread(ClassicOperationExecutor.java:392)
	at com.hazelcast.spi.impl.operationexecutor.classic.ClassicOperationExecutor.runOnCallingThreadIfPossible(ClassicOperationExecutor.java:345)
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.doInvokeLocal(Invocation.java:259)
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.doInvoke(Invocation.java:243)
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.invokeInternal(Invocation.java:210)
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.invoke(Invocation.java:180)
	at com.hazelcast.spi.impl.operationservice.impl.InvocationBuilderImpl.invoke(InvocationBuilderImpl.java:46)
	at com.hazelcast.client.impl.protocol.task.AbstractPartitionMessageTask.processMessage(AbstractPartitionMessageTask.java:65)
	at com.hazelcast.client.impl.protocol.task.AbstractMessageTask.initializeAndProcessMessage(AbstractMessageTask.java:119)
	at com.hazelcast.client.impl.protocol.task.AbstractMessageTask.run(AbstractMessageTask.java:98)
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:127)
	at com.hazelcast.spi.impl.operationexecutor.classic.OperationThread.processPartitionSpecificRunnable(OperationThread.java:159)
	at com.hazelcast.spi.impl.operationexecutor.classic.OperationThread.process(OperationThread.java:142)
	at com.hazelcast.spi.impl.operationexecutor.classic.OperationThread.doRun(OperationThread.java:124)
	at com.hazelcast.spi.impl.operationexecutor.classic.OperationThread.run(OperationThread.java:99)
	at ------ End remote and begin local stack-trace ------.(Unknown Source)
	at com.hazelcast.client.spi.impl.ClientInvocationFuture.resolveResponse(ClientInvocationFuture.java:133)
	at com.hazelcast.client.spi.impl.ClientInvocationFuture.get(ClientInvocationFuture.java:100)
	at com.hazelcast.client.util.ClientDelegatingFuture.get(ClientDelegatingFuture.java:118)
	at com.hazelcast.client.util.ClientDelegatingFuture.get(ClientDelegatingFuture.java:105)
	at com.hazelcast.client.quorum.cache.ClientCacheReadWriteQuorumTest.testGetAsyncOperationSuccessfulWhenQuorumSizeMet(ClientCacheReadWriteQuorumTest.java:292)

```

https://hazelcast-l337.ci.cloudbees.com/view/Official%20Builds/job/Hazelcast-3.x-ZuluJDK7/com.hazelcast$hazelcast-client/81/testReport/junit/com.hazelcast.client.quorum.cache/ClientCacheReadWriteQuorumTest/testGetAsyncOperationSuccessfulWhenQuorumSizeMet/

**Comment by eminn** [comment]
tests are fixed via 06e972beee4a804502925d3133d59a039cba8f33
